node/src/blockchain/blockchain_bridge.rs

```diff
@@ -351,40 +351,43 @@ impl BlockchainBridge {
                 .retrieve_transactions(start_block, end_block, &msg.recipient);
         match retrieved_transactions {
             Ok(transactions) => {
-                debug!(
-                    self.logger,
-                    "Write new start block: {}", transactions.new_start_block
-                );
-                if let Err(e) = self
-                    .persistent_config
-                    .set_start_block(Some(transactions.new_start_block))
-                {
-                    panic! ("Cannot set start block {} in database; payments to you may not be processed: {:?}", transactions.new_start_block, e)
-                };
-                if transactions.transactions.is_empty() {
-                    debug!(self.logger, "No new receivable detected");
+                if let BlockNumber::Number(new_start_block_number) = transactions.new_start_block {
+                    debug!(
+                        self.logger,
+                        "Write new start block: {}",
+                        new_start_block_number.as_u64()
+                    );
+                    if let Err(e) = self
+                        .persistent_config
+                        .set_start_block(Some(new_start_block_number.as_u64()))
```

Comment:
I don't think this is a merge issue. For example, if blockchain_bridge scans 10,000 blocks from 100 to 10,100 and the accountant processes payments received at blocks 4242, and 5758; blockchain_bridge may have persisted that it found all the incoming payments between 100 and 10,100 making it ready to start at 10,101 for the next scan. However, previously if accountant blindly records 4242 then 5758, blockchain_bridge might rescan blocks. By checking to avoid storing a smaller block number, accountant lowers the risk of repeat scans.

I'm reluctant to make such a big change in my card since it's gone on far too long already. I agree with you that it does make sense to record the block number according to what the accountant has successfully recorded. I still think there are some initialization times where we should set and advance the starting block number here in blockchain_bridge while we await our first receivable. If we start and stop the node without receiving any payments the start block number won't advance. When we do restart it will rescan the blocks since the last received payment.

I still argue what the accountant records should be idempotent. It should be safe to reprocess the same transaction block and it can be if we record the sending address, the block (or tx hash), and the amount, we'll never accidentally double count anything. It also would not be harmful to re-scan since it might occur despite our best efforts to prevent it. A database confiscated in a raid won't have any additional information that what is already on the public blockchain -- I agree we are providing a tidy collection of all the 'incriminating evidence' without all the noise. However, the overwhelming benefit is that the books are far less likely to become imbalanced. With the current implementation I suspect frequently declaring bankruptcy will be common.